sunpy/timeseries/sources/goes.py

```diff
@@ -64,63 +64,87 @@ class XRSTimeSeries(GenericTimeSeries):
-        Plots GOES XRS light curve is the usual manner. An example is shown
-        below:
-
-        .. plot::
-
-            import sunpy.timeseries
-            import sunpy.data.sample
-            ts_goes = sunpy.timeseries.TimeSeries(sunpy.data.sample.GOES_XRS_TIMESERIES, source='XRS')
-            ts_goes.peek()
+        Plots the GOES XRS light curve from a pandas dataframe.
 
         Parameters
         ----------
-        title : `str`. optional
+        axes : `matplotlib.axes.Axes`, optional
+            The axes on which to plot the TimeSeries. Defaults to current axes.
+        title : `str`, optional
             The title of the plot. Defaults to "GOES Xray Flux".
         **kwargs : `dict`
             Additional plot keyword arguments that are handed to `~matplotlib.axes.Axes.plot`
             functions.
+
+        Returns
+        -------
+        `~matplotlib.axes.Axes`
+            The plot axes.
         """
-        # Check we have a timeseries valid for plotting
-        self._validate_data_for_plotting()
+        if not axes:
+            axes = plt.gca()
 
-        fig, ax = plt.subplots()
+        self._validate_data_for_plotting()
 
         dates = matplotlib.dates.date2num(parse_time(self.to_dataframe().index).datetime)
-
-        ax.plot_date(dates, self.to_dataframe()['xrsa'], '-',
-                     label=r'0.5--4.0 $\AA$', color='blue', lw=2, **kwargs)
-        ax.plot_date(dates, self.to_dataframe()['xrsb'], '-',
-                     label=r'1.0--8.0 $\AA$', color='red', lw=2, **kwargs)
-
-        ax.set_yscale("log")
-        ax.set_ylim(1e-9, 1e-2)
-        ax.set_title(title)
-        ax.set_ylabel('Watts m$^{-2}$')
-        ax.set_xlabel(datetime.datetime.isoformat(self.to_dataframe().index[0])[0:10])
-
-        ax2 = ax.twinx()
+        axes.plot_date(
+            dates, self.to_dataframe()["xrsa"], "-", label=r"0.5--4.0 $\AA$", color="blue", lw=2, **kwargs
+        )
+        axes.plot_date(
+            dates, self.to_dataframe()["xrsb"], "-", label=r"1.0--8.0 $\AA$", color="red", lw=2, **kwargs
+        )
+
+        axes.set_yscale("log")
+        axes.set_ylim(1e-9, 1e-2)
+        axes.set_title(title)
+        axes.set_ylabel("Watts m$^{-2}$")
+        axes.set_xlabel(datetime.datetime.isoformat(self.to_dataframe().index[0])[0:10])
+
+        ax2 = axes.twinx()
         ax2.set_yscale("log")
         ax2.set_ylim(1e-9, 1e-2)
-        labels = ['A', 'B', 'C', 'M', 'X']
+        labels = ["A", "B", "C", "M", "X"]
         centers = np.logspace(-7.5, -3.5, len(labels))
         ax2.yaxis.set_minor_locator(mticker.FixedLocator(centers))
         ax2.set_yticklabels(labels, minor=True)
         ax2.set_yticklabels([])
 
-        ax.yaxis.grid(True, 'major')
-        ax.xaxis.grid(False, 'major')
-        ax.legend()
+        axes.yaxis.grid(True, "major")
+        axes.xaxis.grid(False, "major")
+        axes.legend()
 
         # TODO: display better tick labels for date range (e.g. 06/01 - 06/05)
```

Comment:
Can we do this?

Reply:
The resultant plots are the same so I suppose we can right